lieer/gmailieer.py

```diff
@@ -549,6 +539,17 @@ def remove_from_list (lst, m):
 
       changed = True
 
+    #limiting the number of messages in the database to local.config.limit parameter
```

Comment:
This might not have the same order as the list from gmail.

Reply:
But, that is the least we can do to make sure that the oldest messages are discarded to maintain the limit. One way might be to chose a different way to sort other than NEWEST_FIRST.

Reply:
Okay, now I see the problem. Maybe, we can go for an oldest thread.

Reply:
I am not sure if there is a reliable way to make notmuch sort messages the same way as gmail does.

Reply:
Yes, but it does come pretty close. The only warning the users should face is that the threads at the very end might be incomplete because of the message limit. But, what we can do is to make sure that threads remain complete and thus remove the old threads as a whole until the sum of the messages is below the limit. What do you prefer for this limit?

Reply:
In the recent commit, I decided to keep the threads intact and thus deciding on the number of messages to be deleted based on threads. And, I also added a progress bar to indicate that older messages are being removed.